packages/taquito/src/prepare/interface.ts

```diff
@@ -0,0 +1,198 @@
+import { OperationContents } from '@taquito/rpc';
+import {
+  BallotParams,
+  DelegateParams,
+  DrainDelegateParams,
+  IncreasePaidStorageParams,
+  OriginateParams,
+  ProposalsParams,
+  RegisterGlobalConstantParams,
+  RevealParams,
+  TransferParams,
+  TransferTicketParams,
+  TxRollupBatchParams,
+  TxRollupOriginateParams,
+  UpdateConsensusKeyParams,
+} from '../operations/types';
+import { ContractMethod } from '../contract/contract-methods/contract-method-flat-param';
+import { ContractMethodObject } from '../contract/contract-methods/contract-method-object-param';
+import { ContractProvider } from '../contract/interface';
+import { ParamsWithKind } from '../operations/types';
+
+/**
+ * @description PrepareProvider is a utility class to output the prepared format of an operation
+ */
+export interface PreparationProvider {
+  /**
+   *
+   * @description Method to prepare a reveal operation
+   *
+   * @param params reveal operation parameters
+   * @param source string or undefined source pkh
```

Comment:
all of these params are a little confusing
is it possible to link the interface thats applicable?

Reply:
Not sure what's confusing about the params, maybe you could elaborate?

Reply:
should param be put here twice? or should it just be referencing what each param is or the interface that it is

Reply:
yes, that's an annotation. you use 1 `@param` annotation per parameter generally

Reply:
it's a pretty common practice, you can see existing examples in `rpc-contract-provider.ts`